orion-lib/src/listeners/http_connection_manager.rs

```diff
@@ -260,6 +267,148 @@ fn per_route_http_filters(
     per_route_filters
 }
 
+/// Applies set_filter_state filters from HCM configuration
+///
+/// This function evaluates format strings and stores the resulting values in the request's
+/// FilterStateExtension (via HTTP extensions) for use in routing decisions and upstream propagation.
+///
+/// # Architecture Note
+///
+/// Filter state is stored per-request using HTTP request extensions, eliminating the need for
+/// global storage and locking. This follows the same pattern as ResponseFlags and EventKind.
+fn apply_set_filter_state_filters<B>(
```

Comment:
Could we move "apply_set_filter_state_filters" and relevant tests to a submod as this file is getting pretty long?